src/Command/Command.php

```diff
@@ -471,28 +471,20 @@ public function queryAll(): array
         return [];
     }
 
-    public function queryColumn(): array|false
+    public function queryColumn(): array
     {
-        /** @psalm-var array<array-key, array<mixed>>|null */
-        $results = $this->queryInternal((int) static::QUERY_MODE_COLUMN);
-
-        if (is_array($results) && count($results)) {
-            return $results;
-        }
+        /** @psalm-var mixed */
+        $results = $this->queryInternal(self::QUERY_MODE_COLUMN);
 
-        return false;
+        return is_array($results) ? $results : [];
     }
 
-    public function queryOne(): array|false
+    public function queryOne(): ?array
     {
-        /** @psalm-var array<array-key, array<mixed>>|null */
-        $results = $this->queryInternal((int) static::QUERY_MODE_ROW);
-
-        if (is_array($results) && count($results)) {
-            return $results;
-        }
+        /** @psalm-var mixed */
+        $results = $this->queryInternal(self::QUERY_MODE_ROW);
 
-        return false;
+        return is_array($results) ? $results : null;
```

Comment:
1. You have error in code - some drivers(pdo and not pdo) may return empty array or null if not found rows (for example - https://www.php.net/manual/ru/mysqli-result.fetch-row.php). 
2. I think current behaviour correct - array or false.
Null it's undefined - but we have "not found data". Not "undefined"

Reply:
How about changes in \Yiisoft\Db\Pgsql\CommandPDO::insertEx (and other usages of queryIne) to behaviour array|false ?
```
        /** @var mixed */
        $queryOne = $this->queryOne();

        return is_array($queryOne) ? $queryOne : false;
```
to
```
        return $this->queryOne();
```

Reply:
Null is "without value" also. In PHP wide use return null when something not found.